modules/default.nix

```diff
@@ -226,7 +237,41 @@ in
       otaUtils # Tools for UEFI capsule updates
     ];
 
+    systemd.tmpfiles.rules = lib.optional nvidiaContainerRuntimeActive "d /var/run/cdi 0755 root root - -";
+
+    systemd.services.nvidia-cdi-generate = {
+      enable = nvidiaContainerRuntimeActive;
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = true;
+        ExecStart =
+          let
+            exe = "${pkgs.nvidia-jetpack.nvidia-ctk}/bin/nvidia-ctk";
+          in
+          toString [
+            exe
+            "--debug"
+            "cdi"
+            "generate"
+            "--nvidia-ctk-path=${exe}" # it is odd that this is needed, should be the same as /proc/self/exe?
+            "--driver-root=${pkgs.nvidia-jetpack.containerDeps}" # the root where nvidia libs will be resolved from
+            "--dev-root=/" # the root where chardevs will be resolved from
+            "--mode=csv"
+            "--csv.file=${pkgs.nvidia-jetpack.l4tCsv}"
+            "--output=/var/run/cdi/jetpack-nixos" # a yaml file extension is added by the nvidia-ctk tool
+          ];
+      };
+      wantedBy = [ "multi-user.target" ];
+    };
+
     # Used by libEGL_nvidia.so.0
     environment.etc."egl/egl_external_platform.d".source = "/run/opengl-driver/share/egl/egl_external_platform.d/";
+
+    # nvidia-ctk fails if /etc/ld.so.cache does not exist. It can be empty, but
```

Comment:
I'm OK with this being in the container, but having it be in the outer NixOS system makes me uncomfortable.

Reply:
This was actually an older assumption that no longer holds with the current status of this PR. It is not needed!